studies/templates/studies/_all_json_and_csv_data.html

```diff
@@ -104,55 +86,46 @@ <h1>All Responses</h1>
                                     the child's birthdate as confirmed in the exit survey.
                                 </p>
                             </div>
-                            <div class="col-md-5 col-sm-6">
-                                <div class="pull-right download-button">
+                            <div class="col">
+                                <div class="text-end my-3 download-button">
                                     Data
-                                    <button id="download-all-data-json"
-                                            class="btn btn-primary"
-                                            type="submit"
-                                            formaction="{% url 'exp:study-responses-download-json' pk=study.id %}"
-                                            {% if not n_responses %} disabled{% endif %}>
-                                        <span> <i class="fa fa-download"></i> (JSON) </span>
-                                    </button>
+                                    {% if n_responses %}
+                                        {% bootstrap_button bs_icon_download|add:" JSON" button_type="submit" button_class="btn btn-primary" id="download-all-data-json" formaction=url_download_all_json %}
+                                    {% else %}
+                                        {% bootstrap_button bs_icon_download|add:" JSON" button_type="submit" button_class="btn btn-primary" id="download-all-data-json" formaction=url_download_all_json disabled="disabled" %}
+                                    {% endif %}
```

Comment:
@okaycj is there a better way to do this? The only thing that differs between the buttons in the if/else blocks is the disabled tag, which used to be set via a conditional (`{% if not n_responses %}...`). But I wasn't sure how to use a conditional here since I'm already inside a template tag. Maybe I could create a variable?

Reply:
We've already spoken about this, but I'll link [the solution here](https://docs.djangoproject.com/en/4.1/ref/templates/builtins/#yesno).

Reply:
@okaycj I tried this and couldn't get it to work 😞 First I tried this:
```
{% bootstrap_button ... disabled=n_responses|yesno:",disabled" %}
```
but that doesn't work because the presence of the disabled attribute itself is what makes the button disabled, so the button is always disabled and there's no string value (empty, false etc.) that will enable it.

I also tried this:
```
{% bootstrap_button ... n_responses|yesno:",disabled='disabled'" %}
```
but that gives me this error:
> 'bootstrap_button' received some positional argument(s) after some keyword argument(s)

I'm stumped and need to move on, so I'm going to leave this as is and merge it. But if you can suggest a way to improve this I will definitely come back and fix it!